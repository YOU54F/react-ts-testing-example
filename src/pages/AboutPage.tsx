import * as React from 'react';

const aboutParagraphs = [
    "An example website written in TypeScript & React.",
    "To demonstrate various testing techniques",
    "such as unit testing with jest",
    "cross-browser compatability testing.",
    "ui testing with cypress."]

export class About extends React.Component<{}, {}> {
    public render() {
        return (
            <div className="about-page-container">
                {
                    aboutParagraphs.map((p,i) => {
                        return (
                            <p 
                            key={i}
                            className="fade-out fade-in" 
                            style={{ animationDelay: (i*0.75)+"s" }} 
                            dangerouslySetInnerHTML={{__html: p}} />
                        )
                    })
                }
            </div>
        )
    }
}