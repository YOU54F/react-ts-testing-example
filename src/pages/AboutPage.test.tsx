import { shallow } from 'enzyme';
import * as React from 'react';
import renderer from 'react-test-renderer';
import { About } from './AboutPage';

describe('About Page', () => {
  it('renders an about paragraph', () => {
    expect(shallow(<About />).find('#paragraphs').length).toEqual(1);
  });
  it('can render the About page', () => {
    const tree = renderer.create(<About />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
