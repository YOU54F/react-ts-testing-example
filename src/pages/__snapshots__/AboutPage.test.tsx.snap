// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About Page can render the About page 1`] = `
<div
  className="about-page-container"
  id="paragraphs"
>
  <p
    className="fade-out fade-in"
    dangerouslySetInnerHTML={
      Object {
        "__html": "An example website written in TypeScript & React.",
      }
    }
    style={
      Object {
        "animationDelay": "0s",
      }
    }
  />
  <p
    className="fade-out fade-in"
    dangerouslySetInnerHTML={
      Object {
        "__html": "To demonstrate various testing techniques",
      }
    }
    style={
      Object {
        "animationDelay": "0.75s",
      }
    }
  />
  <p
    className="fade-out fade-in"
    dangerouslySetInnerHTML={
      Object {
        "__html": "such as unit testing with jest",
      }
    }
    style={
      Object {
        "animationDelay": "1.5s",
      }
    }
  />
  <p
    className="fade-out fade-in"
    dangerouslySetInnerHTML={
      Object {
        "__html": "cross-browser compatability testing.",
      }
    }
    style={
      Object {
        "animationDelay": "2.25s",
      }
    }
  />
  <p
    className="fade-out fade-in"
    dangerouslySetInnerHTML={
      Object {
        "__html": "ui testing with cypress.",
      }
    }
    style={
      Object {
        "animationDelay": "3s",
      }
    }
  />
</div>
`;
